packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2072,6 +2072,19 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         return this.finishNode(nonNullExpression, "TSNonNullExpression");
       }
 
+      let isOptionalCall = false;
+      if (
+        this.match(tt.questionDot) &&
+        this.lookaheadCharCode() === charCodes.lessThan
+      ) {
+        isOptionalCall = true;
```

Comment:
We should mark `state.optionalChainMember` when we see a question dot, which probably starts an optional chain.

For example,
```
a?.<T>().b
```

should be an `OptionalMemberExpression ( OptionalCallExpression, b )`